### DISCUSSION TITLE
Connection by "dhcp_hostname" in ESP32 - Access Point(AP_IF) mode

### DISCUSSION THREAD

**Discussion by nastiliano** [micropython/micropython#9809] [General]
Hello everyone.

I can connect to ESP32 in Station(STA) mode using only the "hostname", not the IP. The "python socket" module resolves the IP automatically from my computer app and it works good.

But in  Access Point(AP_IF) mode it doesn't work:

# Access Point Configuration
    wlan = network.WLAN(network.AP_IF)
    wlan.config(max_clients = 1 , dhcp_hostname = "HOST" , essid = "SSID")
    wlan.active(True)

**Comment by karfas** [comment]
First, you should know what happens in your network and who is responsible for what.

(I assume here you have a simple configuration without dedicated name servers and the like).
- The ESP32 in Station mode.
  - connects to your router. 
  - Your router now knows the name of the ESP32 and it's address (as it was assigned by the router) 
  - Your router has most likely some DNS server built-in or at least answers mDNS queries.
  - Your computer asks the router to resolve a name to an IP.
  - This succeeds, so your application gets an address to connect to.
- The ESP32 in AP mode
  - has no connection to your router
  - A connected client (the computer) **might** ask the AP (depends upon the AP's DHCP configuration) for the name resolution.
  - As the ESP32 AP doesn't provide a DNS server by default, your computer most likely gets no answer (so you need to use the IP address). 

Second, please **read** the _fine_ documentation.

https://docs.micropython.org/en/latest/library/network.WLAN.html does not mention a "dhcp_hostname"; the name given to mDNS is set using the `hostname` parameter.
`dhcp_hostname` is most likely there for backwards compatibility.

Even `hostname` might not work on the ESP32, as the AP_IF doesn't start a DNS server (AFAIK). See https://github.com/espressif/arduino-esp32/issues/5246. 

There are some micropython DNS server implementations (search github). Your mileage may vary - these are 4-5 years old.

**Reply by nastiliano** [reply]
Ok, understood!!!

Thank you very much.

P.D: For me it only works "dhcp_hostname" parameter (I'm using the last micropython version 1.19.1)

**Reply by peterson79** [reply]
For anyone else that lands on this thread.  mDNS doesn't seem to be working on 1.19.1  I just updated to 1.20.0 and without changing any scripts it started working.
